src/Resources/ResourceManager/Implementation/CmdletBase/DeploymentWhatIfCmdletWithParameters.cs

```diff
@@ -0,0 +1,75 @@
+﻿namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation
+{
+    using System.Collections;
+    using System.Management.Automation;
+
+    public abstract class DeploymentWhatIfCmdletWithParameters : DeploymentCmdletBase
```

Comment:
>DeploymentWhatIfCmdletWithParameters [](start = 26, length = 36)

should we just use DeploymentCmdletWithParameters? I know we only support RG/Subscription scope for whatif, but it would be good if we can do it in such a way that we don't need to do breaking changes in the future.

Reply:
I discussed this with @majastrz before and he suggested no to use it. Actually I was reusing it initially - it works fine just for implementation. However if you generate the documentation the SubscriptionAndTenant and ManagementGroup parameter sets will be added, which what-if doesn't support for now.

Reply:
Yeah, deployments work at Tenant, MG, Sub, and RG scopes and what-if only supports Sub and RG scopes for now. What-if will support all the scopes eventually. At that point, we will reuse the Deployments code.

Reply:
sounds good.